src/backend/u64/field.rs

```diff
@@ -448,7 +448,7 @@ impl<'a> ModSqrt for &'a FieldElement {
             let b;
             while i < m {
                 i = i + one;
-                if bool::from(t.pow(&e).ct_eq(&one)) {break;}
+                if t.pow(&e).ct_eq(&one).unwrap_u8() == 1u8 {break;}
                 e = e * two;
             }
 
```

Comment:
This is approved, however, I want to open an issue for using a different modular method (perhaps on associated with the inverse function), to determine positive and negative outputs from the prime_mod_sqrt function.